Source/Core/VideoCommon/FreeLookCamera.cpp

```diff
@@ -194,123 +206,101 @@ class OrbitalController final : public CameraController
 };
 }  // namespace
 
-FreeLookCamera::FreeLookCamera()
-{
-  SetControlType(FreeLook::ControlType::SixAxis);
-}
-
-void FreeLookCamera::SetControlType(FreeLook::ControlType type)
+Common::Vec2 CameraControllerInput::GetFieldOfView() const
 {
-  if (m_current_type && *m_current_type == type)
-  {
-    return;
-  }
-
-  if (type == FreeLook::ControlType::SixAxis)
-  {
-    m_camera_controller = std::make_unique<SixAxisController>();
-  }
-  else if (type == FreeLook::ControlType::Orbital)
-  {
-    m_camera_controller = std::make_unique<OrbitalController>();
-  }
-  else if (type == FreeLook::ControlType::FPS)
-  {
-    m_camera_controller = std::make_unique<FPSController>();
-  }
-
-  m_current_type = type;
+  return Common::Vec2{m_fov_x, m_fov_y};
 }
 
-Common::Matrix44 FreeLookCamera::GetView()
+void CameraControllerInput::DoState(PointerWrap& p)
 {
-  return m_camera_controller->GetView();
+  p.Do(m_speed);
+  p.Do(m_fov_x);
+  p.Do(m_fov_y);
 }
 
-Common::Vec2 FreeLookCamera::GetFieldOfView() const
+void CameraControllerInput::IncreaseFovX(float fov)
 {
-  return Common::Vec2{m_fov_x, m_fov_y};
+  m_fov_x += fov;
+  m_fov_x = std::clamp(m_fov_x, m_min_fov_multiplier, m_fov_x);
```

Comment:
I don't understand why `std::clamp` is being used here.  `clamp` is basically:

```C++
if (m_fov_x < m_min_fov_multiplier)
    return m_min_fov_multiplier;
else if (m_fov_x > m_fov_x)
    return m_fov_x;
else
    return m_fov_x;
```

Wouldn't `std::max(m_fov_x, m_min_fov_multiplier)` be better?  The same applies for `m_fov_y` and `m_speed`.  (I do see that this code existed before this change, but it's still odd).